plugins/net_plugin/net_plugin.cpp

```diff
@@ -4765,9 +4765,18 @@ namespace eosio {
             ++num_peers;
          }
          fc::unique_lock g_conn(c->conn_mtx);
+         if (c->unique_conn_node_id.empty()) { // still connecting, use temp id so that non-connected peers are reported
+            if (!c->p2p_address.empty()) {
+               c->unique_conn_node_id = fc::sha256::hash(c->p2p_address).str().substr(0, 7);
+            } else if (!c->remote_endpoint_ip.empty()) {
+               c->unique_conn_node_id = fc::sha256::hash(c->remote_endpoint_ip).str().substr(0, 7);
+            } else {
+               c->unique_conn_node_id = fc::sha256::hash(std::to_string(c->connection_id)).str().substr(0, 7);
```

Comment:
I thought we could avoid creating an unnecessary temporary 7 character string(), but because of the small string optimization it probably doesn't make any difference, so no need to change.
```suggestion
               c->unique_conn_node_id = std::string_view(fc::sha256::hash(c->p2p_address).str().data(), 7);
            } else if (!c->remote_endpoint_ip.empty()) {
               c->unique_conn_node_id = std::string_view(fc::sha256::hash(c->remote_endpoint_ip).str().data(), 7);
            } else {
               c->unique_conn_node_id = std::string_view(fc::sha256::hash(std::to_string(c->connection_id)).str().data(), 7);
```